### PR TITLE
:bug: Fixed last wave flickering and minor docs update

### LINKED ISSUE #44: The last wave form is always flickering when recording
My code:
```dart
@override
  void initState() {
    super.initState();
    _recorderController = RecorderController();
    _recorderController.record();
  }

...

AudioWaveforms(
    size: const Size(230, 50),
    recorderController: _recorderController,
    waveStyle: const WaveStyle(
      waveColor: Colors.pink,
      spacing: 8.0,
      extendWaveform: true,
      showMiddleLine: false,
   ),
)
```
For some reason the last wave form always appears to be flickering:
recording flickering bug

Is this a bug? or am I doing something wrong?

### PR DESCRIPTION
**Pull request by ujas-m-simformsolutions**
Fixes #44

### CODE REVIEW

**Comment by PassantAbdelatif** [comment]
Any update about this bug that the last wave flickering

**Review by PassantAbdelatif — APPROVED** [review]

**Review by PassantAbdelatif — APPROVED** [review]

**Review by PassantAbdelatif — APPROVED** [review]

**Review by PassantAbdelatif — APPROVED** [review]

**Comment by ujas-m-simformsolutions** [comment]
@PassantAbdelatif This PR still need to be tested for AudioFileWaveforms. That's why it hasn't been merged. We will merge this as soon as possible. Thank you for your patience.

File: lib/src/audio_waveforms.dart [hunk 1]
```diff
@@ -68,12 +68,16 @@ class _AudioWaveformsState extends State<AudioWaveforms> {
...
-              : widget.waveStyle.extraClipperHeight ??
-                  (widget.waveStyle.durationLinesHeight +
-                      (widget.waveStyle.durationStyle.fontSize ??
-                          widget.waveStyle.durationLinesHeight))),
+          clipper: WaveClipper(
+            extraClipperHeight: !widget.waveStyle.showDurationLabel
+                ? 0.0
+                : widget.waveStyle.extraClipperHeight ??
+                    (widget.waveStyle.durationLinesHeight +
+                        (widget.waveStyle.durationStyle.fontSize ??
```
Comment (dhvanit-v-simformsolutions) [inline lib/src/audio_waveforms.dart:76]: What should we do for code readability of this code?

File: lib/src/audio_waveforms.dart [hunk 2]
```diff
@@ -68,12 +68,11 @@ class _AudioWaveformsState extends State<AudioWaveforms> {
...
-          clipper: WaveClipper(!widget.waveStyle.showDurationLabel
-              ? 0.0
-              : widget.waveStyle.extraClipperHeight ??
-                  (widget.waveStyle.durationLinesHeight +
-                      (widget.waveStyle.durationStyle.fontSize ??
-                          widget.waveStyle.durationLinesHeight))),
+          clipper: WaveClipper(
+            extraClipperHeight: _getExtraClipperHeight(),
+            waveWidth:
+                widget.waveStyle.waveThickness + widget.waveStyle.spacing,
```
Comment (aditya-css) [inline lib/src/audio_waveforms.dart:74]: use a getter or late final field for this value.

File: lib/src/audio_waveforms.dart [hunk 3]
```diff
@@ -122,6 +121,19 @@ class _AudioWaveformsState extends State<AudioWaveforms> {
     );
   }
 
+  double _getExtraClipperHeight() {
```
Comment (aditya-css) [inline lib/src/audio_waveforms.dart:124]: try adding comment suggesting the purpose of this method.

**Review by aditya-css — APPROVED** [review]

**Review by guplem — APPROVED** [review]

**Comment by guplem** [comment]
@ujas-m-simformsolutions @dhvanit-v-simformsolutions @vatsaltanna-simformsolutions when will this be merged?
I have been using the latest commit on this pull request as my version for the package for a while now.

In case anyone is interested, this is how I do it:
```yaml
  audio_waveforms: #WAITING FOR UPDATE https://github.com/SimformSolutionsPvtLtd/audio_waveforms/pull/197
    git:
      url: https://github.com/SimformSolutionsPvtLtd/audio_waveforms.git
      ref: d1f8972d8f4d2c957e9077b67cdd9c420774b2a9
```

**Review by vatsaltanna-simformsolutions — APPROVED** [review]

**Comment by guplem** [comment]
@vatsaltanna-simformsolutions Thank you very much for the merge!
Are you planning on creating a new release soon?

**Pull request merged**